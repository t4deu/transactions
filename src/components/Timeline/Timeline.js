// @flow
import React, { Component } from 'react';
import { FlatList } from 'react-native';
import type { Children } from 'react';
import type { Transactions, Transaction } from '../../state/Ducks/Transactions/Types';

import ListItem from './ListItem/ListItem';

import styles from './Timeline.styles';

type Props = {
  header: Children,
  transactions: Transactions,
};

export default class Timeline extends Component<Props> {
  // BUG eslint-plugin-react/#1002
  // eslint-disable-next-line
  renderItem = ({ item, index }: {item: Transaction, index: number}) => (
    <ListItem transaction={item} last={index === this.props.transactions.length - 1} />
  );

  render() {
    return (
      <FlatList
        style={[styles.list]}
        data={this.props.transactions}
        renderItem={this.renderItem}
        keyExtractor={item => item.id}
        ListHeaderComponent={this.props.header}
      />
    );
  }
}
