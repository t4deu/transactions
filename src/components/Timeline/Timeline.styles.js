export default {
  list: {
    flex: 1,
    backgroundColor: 'white',
  },
};
