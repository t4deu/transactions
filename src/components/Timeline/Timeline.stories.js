import React from 'react';
import { storiesOf } from '@storybook/react-native';

import Timeline from './Timeline';

storiesOf('Timeline', module)
.add('with multiple transactions', () => (
  <Timeline
    transactions={[
      { id: 1, type: 'credit', amount: 10.50, description: 'he Beginner Archery and Beginner Crossbow', createdAt: '2018-03-20' },
      { id: 2, type: 'debit', amount: -60, description: 'he Beginner Archery and Beginner Crossbow', createdAt: '13/02/2018' },
      { id: 3, type: 'credit', amount: 115.90, description: 'Crossbow course does not require you to bring any equipment', createdAt: '10:00' },
      { id: 4, type: 'debit', amount: -13.1, description: 'Badminton is a racquet sport played using racquets to hit a shuttlecock across a ne', createdAt: '09:30' },
      { id: 5, type: 'debit', amount: -140, description: 'he Beginner Archery and Beginner Crossbow', createdAt: '2018-03-20' },
      { id: 6, type: 'credit', amount: 10, description: 'teams of eleven players with a spherical ball', createdAt: '2018-03-20' },
    ]}
  />
))
.add('with a single transaction', () => (
  <Timeline
    transactions={[
      { id: 1, type: 'credit', amount: 10, description: 'he Beginner Archery and Beginner Crossbow', createdAt: '2018-03-20' },
    ]}
  />
))
.add('without transactions', () => <Timeline />);
  
