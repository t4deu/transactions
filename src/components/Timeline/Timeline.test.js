import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Timeline from './Timeline';

describe('<Timeline>', () => {
  it('renders empty list', () => {
    const component = shallow(<Timeline />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders with transactions', () => {
    const component = shallow(<Timeline
      transactions={[
          {
            id: 1,
            type: 'credit',
            amount: 10,
            description: 'he Beginner Archery and Beginner Crossbow',
            createdAt: '10:30',
          },
          {
            id: 5,
            type: 'debit',
            amount: -140,
            description: 'he Beginner Archery and Beginner Crossbow',
            createdAt: '10:30',
          },
        ]}
    />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
