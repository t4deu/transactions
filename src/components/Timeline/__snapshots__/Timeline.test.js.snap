// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline> renders empty list 1`] = `
<FlatList
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`<Timeline> renders with transactions 1`] = `
<FlatList
  data={
    Array [
      Object {
        "amount": 10,
        "createdAt": "10:30",
        "description": "he Beginner Archery and Beginner Crossbow",
        "id": 1,
        "type": "credit",
      },
      Object {
        "amount": -140,
        "createdAt": "10:30",
        "description": "he Beginner Archery and Beginner Crossbow",
        "id": 5,
        "type": "debit",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
